addons/mrp/models/mrp_production.py

```diff
@@ -446,8 +462,17 @@ def create(self, values):
         return super(MrpProduction, self).create(values)
 
     def unlink(self):
-        if any(production.state != 'cancel' for production in self):
-            raise UserError(_('Cannot delete a manufacturing order not in cancel state'))
+        if any(production.state == 'done' for production in self):
+            raise UserError(_('Cannot delete a manufacturing order in done state.'))
+        self.action_cancel()
+        productions_done = [p for p in self if p.state != 'cancel']
+        if productions_done:
+            productions_name = ', '.join([prod.display_name for prod in productions_done])
+            raise UserError(_('%s cannot be deleted. Try to cancel them before.') % (productions_name))
```

Comment:
is productions_done a good variable name?
``` python
not_cancel = self.filtered(lambda m: m.state != 'cancel')
if not_cancel:
    raise UserError(_('%s cannot be deleted. Try to cancel them before.') % (not_cancel.mapped('production_name'))
```

Reply:
I kept `productions_name = ', '.join([prod.display_name for prod in not_cancel])` since `not_cancel.mapped('production_name')` return a list and lists are displayed as `['WH/MO/00005', 'WH/MO/00004']` for example :/